pkg/middleware/auth.go

```diff
@@ -90,6 +91,52 @@ func (c JWTVerifier) CheckToken(token string) (*jwt.Token, error) {
 	return tk, nil
 }
 
+// JWTVerifierURL checks token with JWT against JWKS, implements Verifier
+type JWTVerifierURL struct {
+	jwksURL string
+
+	parser *jwt.Parser
+}
+
+// NewJWTVerifierURL creates JWTVerifierURL with JWKS URL
+func NewJWTVerifierURL(jwksURL string) JWTVerifierURL {
+	return JWTVerifierURL{jwksURL: jwksURL, parser: new(jwt.Parser)}
+}
+
+// CheckToken checks the token, validates against JWKS and returns it if it's valid
+func (c JWTVerifierURL) CheckToken(token string) (*jwt.Token, error) {
+	if token == "" {
+		return nil, errors.New("invalid token")
+	}
+
+	tk, err := c.parser.Parse(token, func(token *jwt.Token) (interface{}, error) {
```

Comment:
`func(token *jwt.Token) (interface{}, error)` is really the only differnce between this and the the static JWTVerifier. Consider changing the interface to reflect that abstraction.